ibis/backends/clickhouse/registry.py

```diff
@@ -304,6 +304,8 @@ def _interval_from_integer(translator, expr):
 
 def _literal(translator, expr):
     value = expr.op().value
+    if value is None and expr._dtype.nullable:
```

Comment:
Is the case of `value is None and not expr._dtype.nullable` handled? This could be done in a separate PR if you'd like.

Also, I think you should use the public `type()` API instead of digging around in `_` attributes :)